core/amber/src/main/scala/edu/uci/ics/texera/web/service/WorkflowService.scala

```diff
@@ -153,21 +173,31 @@ class WorkflowService(
         )
       )
     }
+
     var executionID: Long = -1 // for every new execution,
     // reset it so that the value doesn't carry over across executions
     if (WorkflowService.userSystemEnabled) {
-      executionID = ExecutionsMetadataPersistService.insertNewExecution(wId, uidOpt)
+      executionID = ExecutionsMetadataPersistService.insertNewExecution(wId, vId, uidOpt)
```

Comment:
vid = latest vid (store this in the workflow service, and pass it to the `insertNewExecution` and also pass it to the new `workflowContext`